components/homme/src/share/vertremap_base.F90

```diff
@@ -610,26 +611,25 @@ subroutine remap_Q_ppm(Qdp,nx,qsize,dp1,dp2)
           masso(k+1) = masso(k) + ao(k) !Accumulate the old mass. This will simplify the remapping
           ao(k) = ao(k) / dpo(k)        !Divide out the old grid spacing because we want the tracer mixing ratio, not mass.
         enddo
-        !Fill in ghost values. Ignored if vert_remap_q_alg == 2
-        if (vert_remap_q_alg >= 1 .and. vert_remap_q_alg <= 3) then
+        !Fill in ghost values. Ignored if remap_alg == 2
+        if (remap_alg == 1) then
            do k = 1 , gs
-              if (vert_remap_q_alg == 3) then
-                 ao(1   -k) = ao(1)
-                 ao(nlev+k) = ao(nlev)
-              elseif (vert_remap_q_alg == 2 .or. vert_remap_q_alg == 1)then   !Ignored if vert_remap_q_alg == 2
-                 ao(1   -k) = ao(       k)
-                 ao(nlev+k) = ao(nlev+1-k)
-              endif
-            enddo
-         elseif (vert_remap_q_alg == 10) then
-            ext(1) = minval(ao(1:nlev))
-            ext(2) = maxval(ao(1:nlev))
-            call linextrap(dpo(2), dpo(1), dpo(0), dpo(-1), ao(2), ao(1), ao(0), ao(-1), ext(1), ext(2))
-            call linextrap(dpo(nlev-1), dpo(nlev), dpo(nlev+1), dpo(nlev+2),&
-                 ao(nlev-1), ao(nlev), ao(nlev+1), ao(nlev+2), ext(1), ext(2))
-         endif
+              ao(1   -k) = ao(1)
+              ao(nlev+k) = ao(nlev)
+           enddo
+        else if (remap_alg==10) then
+           ext(1) = minval(ao(1:nlev))
+           ext(2) = maxval(ao(1:nlev))
+           call linextrap(dpo(2), dpo(1), dpo(0), dpo(-1), ao(2), ao(1), ao(0), ao(-1), ext(1), ext(2))
+           call linextrap(dpo(nlev-1), dpo(nlev), dpo(nlev+1), dpo(nlev+2),&
+                ao(nlev-1), ao(nlev), ao(nlev+1), ao(nlev+2), ext(1), ext(2))
+        else if (remap_alg==11) then
+           call linextrap(dpo(2), dpo(1), dpo(0), dpo(-1), ao(2), ao(1), ao(0), ao(-1), ext(1), ext(2))
```

Comment:
I think this is calling linextrap with uninit'ed ext(1:2) values. I think you should add a boolean argument `limit` to linextrap saying whether to limit using `ext`.

Reply:
fixed.